### DISCUSSION TITLE
How to contribute UI changes

### DISCUSSION THREAD

**Discussion by ttodua** [AzuraCast/AzuraCast#7412] [Q&A]
in azuracast `frontend` i want to add some missing feature and make PR.
however, in azuracast docs i can't find an instruction how to do that. as i see, docker copies it (instead of mounting a volume) and then builds with vite. therefore making any changes in frontend folder doesn't make sense (would be horrendeous to rebuild docker & vite everytime to see changes).
any simpler way?

**Comment by BusterNeece** [comment]
In the dev version of the installation, we use Vite in `watch` mode, which watches the `frontend` folder and automatically rebuilds and then hot-reloads any changes in the frontend UI. When building for production, we run in `build` mode, which compiles everything down to static CSS/JS assets that are then included in the app.

If you're using a development version of AzuraCast, this file watching system should be set up and working out of the box, and you should see a separate Vite service running in the admin dashboard.

**Reply by ttodua** [reply]
thanks for quick response. I am on Windows and using wsl, and find it a bit tricky to setup dev version. will post a follow up if any issues

**Reply by ttodua** [reply]
I am on windows and it shows me that `/util/docker/common/scripts/my_init` uses CRLF instead of LF (while other `.sh` files are normal). thus causing issues on windows. can you make sure it is LF? 
![image](https://github.com/user-attachments/assets/442db91b-141f-4102-ac3a-96c80b510773)

**Reply by BusterNeece** [reply]
If you're on Windows with WSL, it's strongly recommended to clone the AzuraCast repo inside a WSL filesystem itself (i.e. to spin up another OS like Debian, install it inside there, and then use Docker Desktop's integration with WSL OSes), not just for a significant boost in performance but to avoid line ending issues.

You could also do something like:

```
git config --global core.autocrlf input
```

to change how your Git handles line endings when pulling/pushing files.

**Reply by ttodua** [reply]
thanks for reply!
but the strange thing is that only that one file shows as CRLF, as shown in screenshot, all other files show LF.
if you confirm that file is also LF, then ok, it seems windows glitch, thnx

**Reply by BusterNeece** [reply]
It's because `my_init` has no file extension, so the `.gitattributes` file that we have, which would normally set line endings automatically regardless, doesn't apply to it, causing the Windows specific thing. It's definitely LF in our main repos.

**Reply by ttodua** [reply]
thanks, got it !
